File: api-echo/internal/app/types/watch_history.go
package types

import (
	"time"

	"filmogophery/internal/pkg/constant"
)

type (
	MovieWatchHistory struct {
		ID        int32     `json:"id"`
		Platform  Platform  `json:"platform"`
		WatchedAt time.Time `json:"watchedAt"`
	}

	WatchHistory struct {
		ID        int32         `json:"id"`
		WatchedAt constant.Date `json:"watchedAt"`
		Platform  Platform      `json:"platform"`
		Movie     Movie         `json:"movie"`
	}
)

File: web-next-js/src/interface/watchHistory.ts
import { Movie } from "./movie";
import { Platform } from "./platform";

export type WatchHistory = {
  id: number;
  platform: Platform;
  watchedAt: string;
};

export type MyWatchHistory = {
  id: number;
  platform: Platform;
  watchedAt: string;
  movie: Movie;
};
